modules/custom-status/lib/components/metadata-select-form-token-field.js

```diff
@@ -0,0 +1,123 @@
+import { BaseControl, FormTokenField } from '@wordpress/components';
+import { debounce } from '@wordpress/compose';
+import { useEffect, useMemo, useState } from '@wordpress/element';
+/**
+ * Custom status component
+ * @param object props
+ */
+export default function MetadataSelectFormTokenField( {
```

Comment:
I tried to clean this up by pulling out some common pieces, but it quickly got way too complicated so I've left it as is. I'd like to do this in another PR so it gets the proper attention it needs.